teps/0056-pipelines-in-pipelines.md

```diff
@@ -352,6 +365,335 @@ main-`PipelineRun` to be able to fetch and access the sub-`PipelineRun`'s full s
+    - name: scorecards
+      taskRef:
+        name: scorecards
+    - name: codeql
+      taskRef:
+        name: codeql
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      pipelineRef:
+        name: security-scans
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+The above example can be modified to use `PipelineSpec` instead of `PipelineRef` if the user would prefer to embed the
+`Pipeline` specification:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      pipelineSpec:
+        tasks:
+          - name: scorecards
+            taskRef:
+              name: scorecards
+          - name: codeql
+            taskRef:
+              name: codeql
+    - name: notification
+      taskRef:
+        name: notification
+```
+
+This solution not only separates runtime concerns from authoring time concerns.
+
+These are alternatives to the solution for specification discussed above:
+- [Group `PipelineRef` and `PipelineSpec`](#specification---group-pipelineref-and-pipelinespec)
+- [Use `PipelineRunSpec` in `PipelineTask`](#specification---use-pipelinerunspec-in-pipelinetask)
+- [Reorganize `PipelineTask`](#specification---reorganize-pipelinetask)
+
+### Status
+
+In [TEP-0100][tep-0100] we proposed changes to `PipelineRun` status to reduce the amount of information stored about
+the status of `TaskRuns` and `Runs` to improve performance, reduce memory bloat and improve extensibility. Now that
+those changes have been implemented, the `PipelineRun` status is set up to handle `Pipelines` in `Pipelines` without
+exacerbating the performance and storage issues that were there before.
+
+We can leverage the `PipelineRun.Status.ChildReferences` by populating `ChildReferences` for all child `PipelineRuns`,
+as shown below:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+...
+spec:
+...
+status:
+  conditions:
+    - lastTransitionTime: "2020-05-04T02:19:14Z"
+      reason: Succeeded
+      status: "True"
+      type: Succeeded
+  childReferences:
+    - apiVersion: v1beta1
+      kind: PipelineRun
+      name: sub-pipeline-run
+      conditions:
+        - lastTransitionTime: "2020-05-04T02:10:49Z"
+          reason: Succeeded
+          status: "True"
+          type: Succeeded
+```
+
+The `ChildReferences` will be populated for `Pipelines` in `Pipelines` regardless of the embedded status flags 
+because that is the API behavior we're migrating towards.
+
+These are alternatives to the solution for status discussed above:
+- [Require Minimal Status](#status---require-minimal-status)
+- [Populate Embedded and Minimal Status](#status---populate-embedded-and-minimal-status)
+
+## Design
+
+TODO
```

Comment:
there's still a TODO here

Reply:
yes, in this pr we're focused on the API change proposal - the design details of how it interacts with other features will be added in a subsequent PR (the TEP is still in proposed state)

Reply:
the above is mentioned in the first sentence in the proposal

Reply:
sorry, long day :(